### PR TITLE
chore: refactor the snapshot code

### LINKED ISSUE #409: refactor the snapshot code
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

I found `extractSnapshotInfo` func is not in good coding style:
 - I think we should remove `if !strings.Contains(snapshotID, "/subscriptions/") {` condition since it's not `extractSnapshotInfo`
 - add a unit test for `extractSnapshotInfo` func and comments for the format of  `extractSnapshotInfo` func 

https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/873e575520dfe9efa4686a3ab90845f6bd2ebb85/pkg/azuredisk/controllerserver.go#L795-L810

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

- `extractSnapshotInfo` func is not in good coding style. Remove the condition that checking whether the id contains `/subscriptions/`.
- Delete the unnecessary unit test case.

**Which issue(s) this PR fixes**:

Fixes #409

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/azuredisk/controllerserver.go [hunk 1]
```diff
@@ -816,19 +837,13 @@ func isCSISnapshotReady(state string) (bool, error) {
...
-		resourceGroup = d.cloud.ResourceGroup
-	} else {
-		if snapshotName, err = getSnapshotName(snapshotID); err != nil {
-			return "", "", err
-		}
-		if resourceGroup, err = GetResourceGroupFromURI(snapshotID); err != nil {
-			return "", "", err
-		}
+// The format of snapshot id is /subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Compute/snapshots/snapshot-xxx-xxx.
+func (d *Driver) extractSnapshotInfo(snapshotID string) (snapshotName, resourceGroup string, err error) {
```
Comment (andyzhangx) [inline pkg/azuredisk/controllerserver.go:841]: `extractSnapshotInfo` --> `gettSnapshotInfo`

File: pkg/azuredisk/controllerserver.go [hunk 2]
```diff
@@ -648,13 +648,24 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
...
 
-	snapshotName, resourceGroup, err := d.extractSnapshotInfo(req.SnapshotId)
-	if err != nil {
-		return nil, err
+	var (
+		snapshotName, resourceGroup string
+		err                         error
+	)
+
+	if !strings.Contains(snapshotID, "/subscriptions/") {
```
Comment (andyzhangx) [inline pkg/azuredisk/controllerserver.go:661]: `strings.Contains(strings.ToLower(snapshotID), "/subscriptions/")`

Comment (andyzhangx) [reply pkg/azuredisk/controllerserver.go:661]:
```
if strings.Contains(strings.ToLower(snapshotID), "/subscriptions/") {
		snapshotName, resourceGroup, err = d.extractSnapshotInfo(snapshotID)
		if err != nil {
			return nil, err
		}
} else {
		snapshotName = snapshotID
		resourceGroup = d.cloud.ResourceGroup
}
```

File: pkg/azuredisk/controllerserver.go [hunk 3]
```diff
@@ -701,9 +712,19 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
...
 func (d *Driver) getSnapshotByID(ctx context.Context, snapshotID, sourceVolumeID string) (*csi.Snapshot, error) {
-	snapshotName, resourceGroup, err := d.extractSnapshotInfo(snapshotID)
-	if err != nil {
-		return nil, err
+	var (
+		snapshotName, resourceGroup string
+		err                         error
+	)
+
+	if !strings.Contains(snapshotID, "/subscriptions/") {
```
Comment (andyzhangx) [inline pkg/azuredisk/controllerserver.go:720]:
```
if strings.Contains(strings.ToLower(snapshotID), "/subscriptions/") {
		snapshotName, resourceGroup, err = d.extractSnapshotInfo(snapshotID)
		if err != nil {
			return nil, err
		}
} else {
		snapshotName = snapshotID
		resourceGroup = d.cloud.ResourceGroup
}
```

File: pkg/azuredisk/controllerserver.go (hunk 1 above)
Comment (ZeroMagic) [reply pkg/azuredisk/controllerserver.go]: done

File: pkg/azuredisk/controllerserver.go (hunk 2 above)
Comment (ZeroMagic) [reply pkg/azuredisk/controllerserver.go]: done

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**